beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -221,25 +215,95 @@ func (s *Server) GetBlockAttestations(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
-	root, err := blk.Block().HashTreeRoot()
-	if err != nil {
-		httputil.HandleError(w, "Could not get block root: "+err.Error(), http.StatusInternalServerError)
-		return
+	resp := &structs.GetBlockAttestationsResponse{
+		Data:                atts,
+		ExecutionOptimistic: isOptimistic,
+		Finalized:           s.FinalizationFetcher.IsFinalized(ctx, root),
 	}
-	isOptimistic, err := s.OptimisticModeFetcher.IsOptimisticForRoot(ctx, root)
-	if err != nil {
-		httputil.HandleError(w, "Could not check if block is optimistic: "+err.Error(), http.StatusInternalServerError)
+	httputil.WriteJson(w, resp)
+}
+
+// GetBlockAttestationsV2 retrieves attestation included in requested block.
+func (s *Server) GetBlockAttestationsV2(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetBlockAttestationsV2")
+	defer span.End()
+
+	blk, isOptimistic, root := s.blockData(ctx, w, r)
+	if blk == nil {
 		return
 	}
+	consensusAtts := blk.Block().Body().Attestations()
 
-	resp := &structs.GetBlockAttestationsResponse{
-		Data:                atts,
+	atts := make([]json.RawMessage, len(consensusAtts))
+	v := blk.Block().Version()
+
+	if v >= version.Electra {
+		for i, att := range consensusAtts {
+			a, ok := att.(*eth.AttestationElectra)
+			if ok {
```

Comment:
similar feedback to another PR we can avoid the else statement by reversing this